lib/livebook/content_loader.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Livebook.ContentLoader do
+  @moduledoc false
+
+  @doc """
+  Rewrite known URLs, so that they point to plain text file rather than HTML.
+
+  Currently the rewerites handle:
+
+    * GitHub files
+    * Gist files
+  """
+  @spec rewrite_url(String.t()) :: String.t()
+  def rewrite_url(url) do
+    url
+    |> URI.parse()
+    |> do_rewrite_url()
+    |> URI.to_string()
+  end
+
+  defp do_rewrite_url(%URI{host: "github.com"} = uri) do
+    case String.split(uri.path, "/") do
+      ["", owner, repo, "blob", hash | file_path] ->
+        path = Enum.join(["", owner, repo, hash | file_path], "/")
+
+        %{
+          uri
+          | path: path,
+            host: "raw.githubusercontent.com",
+            authority: "raw.githubusercontent.com"
+        }
+
+      _ ->
+        uri
+    end
+  end
+
+  defp do_rewrite_url(%URI{host: "gist.github.com"} = uri) do
+    case String.split(uri.path, "/") do
+      ["", owner, hash] ->
+        path = Enum.join(["", owner, hash, "raw"], "/")
+
+        %{
+          uri
+          | path: path,
+            host: "gist.githubusercontent.com",
+            authority: "gist.githubusercontent.com"
+        }
+
+      _ ->
+        uri
+    end
+  end
+
+  defp do_rewrite_url(uri), do: uri
+
+  @doc """
+  Loads binary content from the given URl and validates if its plain text.
+  """
+  @spec fetch_content(String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  def fetch_content(url) do
+    case :httpc.request(:get, {url, []}, [], body_format: :binary) do
```

Comment:
Unfortunately httpc does not verify SSL headers by default. We should probably add the :castors dependency and add the magical combination of flags here. @ericmj or @wojtekmach or @whatyouhide, do you know the magic combination to make httpc safe? Alternatively we can use Mint to start a connection temporarily and then tear it down.

Reply:
Thanks for the pointer, that's been useful! I also found some docs on that from the EEF Security WG [here](https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/inets), so I think we need just a few options to make things secure. I created a notebook ([gist](https://gist.github.com/jonatanklosko/5e20ca84127f6b31bbe3906498e1a1d7)) with my research and understanding of things, please let me know if I missed anything.